ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/ClassLoadInvoker.java

```diff
@@ -599,6 +635,22 @@ public List<String> availableModuleDeclarations() {
         return moduleDclnStrings;
     }
 
+    @Override
+    public List<String> newVariableNames() {
+        return newDefinedVariableNames;
+    }
+
+    @Override
+    public List<String> newModuleDeclarations() {
+        return newModuleDeclnNames;
+    }
+
+    @Override
+    public void clearPreviousVariablesAndModuleDclnsNames() {
+        newDefinedVariableNames = new ArrayList<>();
+        newModuleDeclnNames = new ArrayList<>();
+    }
```

Comment:
Can't we use `.clear()` instead of initializing variables again ?

Reply:
Updated [fb70b93](https://github.com/ballerina-platform/ballerina-lang/pull/35871/commits/fb70b9385e531f1e1ba4952c5bde77c1899b6fd7).